### ISSUE TITLE
LogClient logic: LogClient.AddLeaf always fails on WaitForInclusion

### ISSUE THREAD

**Issue by Liamsi** [google/trillian#728]
TL;DR: `signature is nil` happens because verification is called on an empty signed log root `DeadlineExceeded` happens because the client keeps retrying but doesn't receive an inclusion proof for the leaf it's waiting for (even if the first is fixed). The latter is an error in transparency.

1) If `LogClient.AddLeaf` is called the first time (on a freshly created tree) it returns
`WaitForInclusion(): signature is nil `

2) In consecutive calls it times out with:
`WaitForInclusion(): rpc error: code = DeadlineExceeded desc = context deadline exceeded`

The first error message 1. happens on the initial `UpdateRoot` [here](https://github.com/google/trillian/blob/0ae8e4caacf2e04d87d9a437414c00ba720109d6/client/client.go#L198-L199). 
`UpdateRoot` trys to `GetLatestSignedLogRoot` and receives an "empty" response  (`{TimestampNanos:0 RootHash:[] TreeSize:0 Signature:<nil> LogId:0 TreeRevision:0}`) on which it then calls `VerifyRoot` / `crypto/verifier.Verify`.  I think the `if c.root.TreeSize == 0 ` part that waits for an RootUpdate to happen (in `WaitForInclusion`) should happen before calling `UpdateRoot`.

The second error happens because the client keeps retrying and the deadline exceeds before it sees a proper inclusion proof (`No leaves for hash: yourfavouritehashersgibberish84deadbeefetc`). Potenially the hasher in the backend and the `LogClient` don't match.
